kuma/javascript/src/header/main-menu.jsx

```diff
@@ -0,0 +1,266 @@
+            let currentTarget = event.target;
+            if (
+                currentTarget instanceof HTMLButtonElement &&
+                currentTarget.classList.contains('top-level-entry')
+            ) {
+                if (currentTarget.nextElementSibling) {
+                    currentTarget.nextElementSibling.setAttribute(
+                        'aria-hidden',
+                        'false'
+                    );
+                }
+            }
+        });
+    }
+
+    /**
+     * Handles touch events on mobile devices
+     * @param {Object} menu - The current HTML nav element
+     */
+    function mobileInteractionHandlers(menu) {
+        menu.addEventListener('touchend', (event: TouchEvent) => {
+            event.stopImmediatePropagation();
+            let currentTarget = event.target;
+            if (
+                currentTarget instanceof HTMLButtonElement &&
+                currentTarget.classList.contains('top-level-entry')
+            ) {
+                let subMenu = currentTarget.nextElementSibling;
+
+                if (subMenu) {
+                    let currentAriaState = subMenu.getAttribute('aria-hidden');
+                    let newAriaState =
+                        currentAriaState === 'true' ? 'false' : 'true';
+                    subMenu.setAttribute('aria-hidden', newAriaState);
+                    subMenu.classList.toggle('show');
+                }
+            }
+        });
+    }
+
+    /**
+     * Hide the visible submenu in main navigation
+     */
+    function hideSubMenus() {
+        let visibleSubMenu = document.querySelector(
+            'li ul[aria-hidden="false"]'
+        );
+        if (visibleSubMenu) {
+            visibleSubMenu.setAttribute('aria-hidden', 'true');
+            visibleSubMenu.classList.toggle('show');
+        }
+    }
+
+    useEffect(() => {
+        let menu = document.getElementById('header-main-nav');
+        var mediaQuery = window.matchMedia('(min-width: 47.9375em)');
+        if (menu) {
+            if (mediaQuery.matches) {
+                desktopInteractionHandlers(menu);
+            } else {
+                /* Because the header is not rerendered during client-side
+                   navigation, we call this function to ensure the currrently
+                   visible submenu is hidden when a link it triggered */
+                hideSubMenus();
+                mobileInteractionHandlers(menu);
+            }
+        }
+    });
+
+    // One of the menu items has a URL that we need to substitute
+    // the current mdnDocument path into. Compute that now.
+    let path = encodeURIComponent(
+        `/${locale}` + (mdnDocument ? `/docs/${mdnDocument.slug}` : '')
+    );
+
+    return (
+        <nav id="header-main-nav" className="main-nav" role="navigation">
+            <ul>
+                {menus.map((menuEntry, index) => (
+                    <li key={index} className="top-level-entry-container">
```

Comment:
This makes me nervous. The `key` just needs to be something unique. What's guaranteed to be unique is the `menuEntry.url`.

Reply:
Grr.. My bad. An `menuEntry` doesn't have a `.url`. That's what `menuEntry[0]` has. So this line should instead be `key={menuEntry.label}`.